### PR TITLE
Send click envent on search to algolia

### PR DESCRIPTION
**Pull request by alachambre**
## Behavior

Use the Algolia Insight API to send click event when a search suggestion is selected, in order to have more data to monitor about the efficiency of the search. 

When a user click on a search suggestion, an HTTP request is sent to `https://insights.algolia.io/1/events`

If the request is correctly formatted, then the following success response is expected: 
`{"status":200,"message":"OK"}`

Then algolia will process the click event to add it to our analytics dashboard. IT takes time! (hours). 

## Implementation details
I had to dig a bit to understand how to set up the insight feature with the docsearch layer. 

Here is the API to use: https://www.algolia.com/doc/rest-api/insights/
We are using the docsearch dropdown, which add a layer between us and the search: https://docsearch.algolia.com/docs/behavior

To enable the search we have to instantiate a `docsearch` object with some params (apiKey, id's...). The constructor also accept some methods to implement custom behavior, which is only partially and most of the time poorly documented. 
We are using the methods `handleSelected` and `queryDataCallback`. 

The first one allows us manage the logic just after the user click on a search suggestion. Send a click event to Algolia and redirect the user to the url he clicked on. 

The second one allows us to retrieve the response of the initial search request. This response contains the queryID and the raw hits. Each hit contains an object ID. all those ID's are required to send a proper click event. 

So basically we use the search response to store locally some id's, and if the user click on a suggestion we use those id's to send a click event. 

To get the ID's in the search response I had to add the docsearch option `clickAnalytics: true` (information perfectly hidden at the bottom of a random documentation page :) )

### CODE REVIEW

File: src/js/06-docsearch.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  for (let i = 0; i < hits.length; i++) {
+    var hit = hits[i]
+    if (hit.url === suggestion.url) {
+      position = i + 1
+      objectID = hit.objectID
+      break
+    }
+  }
+
+  // Seems that Gulp uglify is unable to deal with proper js multiline String...
```
Comment (tbouffard) [inline src/js/06-docsearch.js:26]: 😒

Comment (alachambre) [reply src/js/06-docsearch.js:26]: yup, didn't manage to make it work correctly, so I went for the old fashioned way :)
